maestro/deploy/openshift-templates/namespace-template.yml

```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: v1
+kind: Template
+metadata:
+  name: maestro-ns
+parameters:
+
+  - name: NAMESPACE
+    description: The OpenShift Namespace where the resources will be created.
+    displayName: Namespace
+    required: true
+    value: maestro
+
+  - name: ISTIO_VERSION
+    description: The version of Istio to use for this namespace
+    displayName: Istio Version
+    required: true
+
+objects:
+
+  - apiVersion: v1
+    kind: Namespace
+    metadata:
+      name: ${NAMESPACE}
+      labels:
+        istio.io/rev: ${ISTIO_VERSION}
```

Comment:
I'm not sure if we want to "manage" this istio version label on namespaces in the future because during an istio upgrade we would want to make sure this isn't getting applied to the old version, but lgtm for now

Reply:
i agree. istio upgrads and sidecar injection annotations need to be coordinated eventually if we decide to go with istio.